pkg/driver_cryptocell_310/psa_cryptocell_310/cipher_chacha20.c

```diff
@@ -0,0 +1,259 @@
+/*
+ * Copyright (C) 2024 TU Dresden
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     pkg_driver_cryptocell_310
+ * @{
+ *
+ * @file
+ * @brief       PSA Crypto wrapper for the CryptoCell 310 ChaCha API.
+ *
+ * @author      Lennard Melling <lennard.melling@msx.tu-dresden.de>
+ *
+ */
+#ifndef PSA_CRYPTOCELL_310_CHACHA_H
+#define PSA_CRYPTOCELL_310_CHACHA_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include "psa/crypto.h"
+#include "crys_chacha.h"
+#include "cryptocell_310_util.h"
+#include "psa_error.h"
+#include "unaligned.h"
+
+#define ENABLE_DEBUG    0
+#include "debug.h"
+
+psa_status_t _setup(psa_cipher_chacha20_ctx_t *ctx,
+                    uint8_t *key_data,
+                    size_t key_length,
+                    CRYS_CHACHA_EncryptMode_t mode)
+{
+    DEBUG("Peripheral ChaCha20 Cipher setup");
+    psa_status_t status = PSA_ERROR_CORRUPTION_DETECTED;
+
+    if (key_length != CRYS_CHACHA_KEY_MAX_SIZE_IN_BYTES)
+    {
+        return PSA_ERROR_INVALID_ARGUMENT;
+    }
+    ctx->is_finished = 0;
+    ctx->ctx.pre_setup.mode = mode;
+    ctx->ctx.pre_setup.key = key_data;
+    status = PSA_SUCCESS;
+    return status;
```

Comment:
I adopted the convention from other places, where we at the start of the function define status = PSA_ERROR_CORRUPTION_DETECTED, so that if in any way the code between this and the return statement is not run, then we can see there was a corruption.

Reply:
Hum, sure in other places where we have a chain of calls that return `psa_status_t` it makes sense to have a local variable that should be initialized to something, but how would some code not be run? Are you testing for compiler bugs? :P

Reply:
I thought that that is the reason, why it is used in other cases, but you are right, in this case it is probably better to just return the value.